plugins/outputs/iotdb/iotdb.go

```diff
@@ -0,0 +1,277 @@
+	DeviceIDList     []string
+	MeasurementsList [][]string
+	ValuesList       [][]interface{}
+	DataTypesList    [][]client.TSDataType
+	TimestampList    []int64
+	// extra tags
+	TagsList [][]*telegraf.Tag
+}
+
+func (*IoTDB) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init is for setup, and validating config.
+func (s *IoTDB) Init() error {
+	if time.Duration(s.Timeout)/time.Millisecond < 0 {
+		return errors.New("negative timeout")
+	}
+	if !choice.Contains(s.ConvertUint64To, []string{"int64", "int64_clip", "text"}) {
+		return fmt.Errorf("unknown 'uint64_conversion' method %q", s.ConvertUint64To)
+	}
+	if !choice.Contains(s.TimeStampUnit, []string{"second", "millisecond", "microsecond", "nanosecond"}) {
+		return fmt.Errorf("unknown 'timestamp_precision' method %q", s.TimeStampUnit)
+	}
+	if !choice.Contains(s.TreatTagsAs, []string{"fields", "device_id"}) {
+		return fmt.Errorf("unknown 'convert_tags_to' method %q", s.TreatTagsAs)
+	}
+	s.Log.Info("IoTDB output plugin initialization completed.")
+	return nil
+}
+
+func (s *IoTDB) Connect() error {
+	sessionConf := &client.Config{
+		Host:     s.Host,
+		Port:     s.Port,
+		UserName: s.User,
+		Password: s.Password,
+	}
+	var ss = client.NewSession(sessionConf)
+	s.session = &ss
+	if err := s.session.Open(false, int(time.Duration(s.Timeout)/time.Millisecond)); err != nil {
+		return fmt.Errorf("connecting to %s:%s failed: %w", s.Host, s.Port, err)
+	}
+	return nil
+}
+
+func (s *IoTDB) Close() error {
+	// Close any connections here.
+	// Write will not be called once Close is called, so there is no need to synchronize.
+	_, err := s.session.Close()
+	return err
+}
+
+// Write should write immediately to the output, and not buffer writes
+// (Telegraf manages the buffer for you). Returning an error will fail this
+// batch of writes and the entire batch will be retried automatically.
+func (s *IoTDB) Write(metrics []telegraf.Metric) error {
+	// Convert Metrics to Records with Tags
+	rwt, err := s.ConvertMetricsToRecordsWithTags(metrics)
+	if err != nil {
+		return err
+	}
+	// Write to client.
+	// If first writing fails, the client will automatically retry three times. If all fail, it returns an error.
+	if err := s.WriteRecordsWithTags(rwt); err != nil {
+		s.Log.Errorf("IoTDB Write Error: %s", err.Error())
+		return err
+	}
+	return nil
+}
+
+// Find out data type of the value and return it's id in TSDataType, and convert it if necessary.
+func (s *IoTDB) getDataTypeAndValue(value interface{}) (client.TSDataType, interface{}) {
+	switch v := value.(type) {
+	case int32:
+		return client.INT32, v
+	case int64:
+		return client.INT64, v
+	case uint32:
+		return client.INT64, int64(v)
+	case uint64:
+		switch s.ConvertUint64To {
+		case "int64_clip":
+			if v <= uint64(math.MaxInt64) {
+				return client.INT64, int64(v)
+			}
+			return client.INT64, int64(math.MaxInt64)
+		case "int64":
+			return client.INT64, int64(v)
+		case "text":
+			return client.TEXT, fmt.Sprintf("%d", v)
```

Comment:
Can we please do an explicit conversion instead of relying on Sprintf to do the right thing!?
```suggestion
			return client.TEXT, strconv.FormatUint(v, 10)
```

Reply:
This expression is really better. Thanks for your suggestion!